### PR TITLE
Assignment12

### PR DESCRIPTION
**Pull request by prajjwalkumarpanzade**
Added test cases to all controllers and model using Factorybot and Faker

### CODE REVIEW

File: Assignment12/spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      end
+    end
+  end
+
+  describe "GET #show" do
+    let!(:user) { create(:user) }
+
+    it "returns a success response" do
+      get :show, params: { id: user.id }
+      expect(response).to be_successful
```
Comment (samu80) [inline Assignment12/spec/requests/users_spec.rb:53]: Add line space after API call

File: Assignment12/spec/requests/books_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      it "updates the requested book" do
+        put :update, params: { id: book.id, book: { title: "Updated Title" } }
+        book.reload
+        expect(book.title).to eq("Updated Title")
+      end
+
+      it "renders a JSON response with the book" do
+        put :update, params: { id: book.id, book: { title: "Updated Title" } }
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
```
Comment (samu80) [inline Assignment12/spec/requests/books_spec.rb:49]: You can check book id is correct or not

File: Assignment12/spec/requests/users_spec.rb (hunk 1 above)
Comment (prajjwalkumarpanzade) [reply Assignment12/spec/requests/users_spec.rb:53]: ok

File: Assignment12/spec/requests/books_spec.rb (hunk 2 above)
Comment (prajjwalkumarpanzade) [reply Assignment12/spec/requests/books_spec.rb:49]: ok

File: Assignment12/spec/requests/books_spec.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    it "destroys the requested book" do
+      expect {
+        delete :destroy, params: { id: book.id }
+      }.to change(Book, :count).by(-1)
+    end
+
+    it "renders a JSON response with a success message" do
+      delete :destroy, params: { id: book.id }
+      expect(response).to have_http_status(:ok)
```
Comment (samu80) [inline Assignment12/spec/requests/books_spec.rb:77]: Add line space between every API call and expected statements

Comment (prajjwalkumarpanzade) [reply Assignment12/spec/requests/books_spec.rb:77]: done

File: Assignment12/spec/requests/books_spec.rb [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    end
+  end
+
+  describe "DELETE #destroy" do
+    let!(:book) { create(:book) }
+
+    it "destroys the requested book" do
+      expect {
+        delete :destroy, params: { id: book.id }
+
```
Comment (samu80) [inline Assignment12/spec/requests/books_spec.rb]: Here line space is not required

Comment (prajjwalkumarpanzade) [reply Assignment12/spec/requests/books_spec.rb]: done

**Review by samu80 — APPROVED** [review]

**Pull request merged**